WickedEngine/wiHelper.cpp

```diff
@@ -1410,8 +1415,12 @@ namespace wi::helper
 			WideCharToMultiByte(CP_UTF8, 0, from.c_str(), -1, &to[0], num, NULL, NULL);
 		}
 #else
-		std::wstring_convert<std::codecvt_utf8<wchar_t>> cv;
-		to = cv.to_bytes(from);
+		std::mbstate_t state = std::mbstate_t {};
+		const wchar_t* from_c = from.c_str();
+		const std::size_t len = 1 + std::wcsrtombs(nullptr, &from_c, 0, &state);
+		std::vector<char> to_c(len);
```

Comment:
it can directly write into the `to.data()` instead of using an extra std::vector and copy.

Reply:
but that it's a problem if the to buffer is not sized or intialized.